lib/Remote/RemoteWebElement.php

```diff
@@ -326,14 +326,21 @@ public function sendKeys($value)
     {
         $local_file = $this->fileDetector->getLocalFile($value);
 
+        $params = [];
         if ($local_file === null) {
             if ($this->isW3cCompliant) {
-                $params = [
-                    'text' => WebDriverKeys::encode($value, true),
-                    ':id' => $this->id,
-                ];
+                // Work around the Geckodriver NULL issue by splitting on NULL and calling sendKeys multiple times.
+                // See https://bugzilla.mozilla.org/show_bug.cgi?id=1494661.
+                $encodedvalues = WebDriverKeys::encode($value, true);
+                $values = explode(WebDriverKeys::NULL, $encodedvalues);
+                foreach ($values as $encodedvalue) {
+                    $params[] = [
+                        'text' => $encodedvalue,
```

Comment:
Maybe here the single-use variable is not necessary:

```suggestion
                $encodedValues = explode(WebDriverKeys::NULL, WebDriverKeys::encode($value, true));
                foreach ($encodedValues as $encodedValue) {
                    $params[] = [
                        'text' => $encodedValue,
```